src/torrent.cpp

```diff
@@ -4166,8 +4166,7 @@ namespace {
 	// announce it to peers ahead of time to eliminate the
 	// round-trip times involved in announcing it, requesting it
 	// and sending it
-	// TODO: 2 use chrono type for time duration
-	void torrent::predicted_have_piece(piece_index_t const index, int const milliseconds)
+	void torrent::predicted_have_piece(piece_index_t const index, time_duration const duration)
```

Comment:
I'm a bit torn here. On one hand, using the "generic" `time_duration` here is nice, because it hides internals. But on the other hand, it does mean the `milliseconds32` will be converted to nanoseconds (or maybe microseconds) in an `in64_t` just to be converted back again.

If this function would take a `milliseconds32` directly, you would still have the type-safety, you would just leak to the caller what format you're using internally in the function

Reply:
I'm not too sure about the internals and I'm not at my workstation so I don't have the code in front of me - but if we make the `threshold` a time_duration, then we can use the regular `milliseconds` type I think? I chose `milliseconds32` to be able to use `eta.count()` without a static_cast to int.

If that works, we would keep the time_duration in `torrent::predicted_have_piece` but won't have to to the conversion from `milliseconds32`.

That said, I'm not well-versed in these APIs :)

Reply:
> but if we make the threshold a time_duration, then we can use the regular milliseconds type I think? I chose milliseconds32 to be able to use eta.count() without a static_cast to int.

That sounds right. I thought you picked `milliseconds32` to maintain the `int` width in the existing code :)